Web DLP class/methods

daxm opened the pull request:
Updated my "main" with Mitch's.  Also merged in my web_dlp branch.

mitchos commented on pyzscaler/zia/__init__.py:
```diff
@@ -210,3 +211,19 @@ def vips(self):
         """
         return DataCenterVIPSAPI(self)
+
+    @property
```
I think you may have accidentally duplicated this class attribute. We have the admin_and_role_management attribute defined on line 95. 

Can you just remove lines 216->221 when you resubmit?

mitchos commented on pyzscaler/zia/admin_and_role_management.py:
```diff
@@ -8,15 +8,13 @@ class AdminAndRoleManagementAPI(APIEndpoint):
     def add_user(self, name: str, login_name: str, email: str, password: str, **kwargs) -> Box:
```
I'm not sure what was changed in this file, just looks like a blank line was removed somewhere. I think we can remove this from the pull request.

mitchos commented on pyzscaler/zia/web_dlp.py:
```diff
@@ -0,0 +1,82 @@
+
+
+class WebDLP(APIEndpoint):
+    def get_all(self, **kwargs) -> BoxList:
```
Are you happy if we rename these methods to align with the rest of the SDK? 

I've taken the approach that Zscaler may add additional objects to some of these API endpoints, which would mean we need to be a bit more verbose about the verbs and what we're doing. So at the moment it appears that the web_dlp endpoint just has policy rules, but if they were to add anything else then just having `zia.web_dlp.get_all() `would be ambiguous.

So while it may seem a bit over the top, having a descriptive verb/noun for the methods ensures we don't need to go back and refactor if the API endpoints are expanded. You should see a _module.attribute.verb_noun_ convention used across the SDK.

If we're returning a list of items then we should expect something like `zia.web_dlp.list_rules()`

mitchos commented on pyzscaler/zia/web_dlp.py:
```diff
@@ -0,0 +1,82 @@
+        """
+        return self._get("webDlpRules")
+
+    def get_item(self, item_id: str) -> Box:
```
Consider naming to `get_rule()`

mitchos commented on pyzscaler/zia/web_dlp.py:
```diff
@@ -0,0 +1,82 @@
+        """
+        return self._get("webDlpRules/lite")
+
+    def post(self, payload: json) -> Box:
```
Consider naming to `add_rule()`

mitchos commented on pyzscaler/zia/web_dlp.py:
```diff
@@ -0,0 +1,82 @@
+        """
+        return self._post("webDlpRules", json=payload)
+
+    def put(self, item_id: str, payload: json) -> Box:
```
Consider naming `update_rule()`

mitchos commented on pyzscaler/zia/web_dlp.py:
```diff
@@ -0,0 +1,82 @@
+        """
+        return self._put(f"webDlpRules/{item_id}", json=payload)
+
+    def delete(self, item_id: str) -> requests.Response:
```
Consider naming `delete_rule()`

mitchos replied:
If we return the status code, just a reminder that the return type is `Int`.

mitchos commented on pyzscaler/zia/web_dlp.py:
```diff
@@ -0,0 +1,82 @@
+        Returns:
+                Requests.Response object.
+        """
+        return self._delete(f"webDlpRules/{item_id}")
```
Often I've found ZIA responses to DELETE to be inconsistent. I've opted to just return the status code instead, which gives us at least a uniform response to expect on a successful DELETE.

I'd recommend:
return `self._delete(f"webDlpRules/{item_id}").status_code`

mitchos reviewed (changes requested):
Thanks for getting started on this @daxm, really appreciate it! 

After addressing those review points we can merge into develop and add some additional work to round it off:

- Docstrings complete with examples
- Adding the docsrc RST files to ensure our readthedocs site will pick up the changes
- Test coverage

Overall great quality pull request and thankyou thankyou thankyou!

daxm commented:
I appreciate your detailed response.  Let me try to give context to your notes:
  - Removed __init__.py duplication.  This was a hangover from when I was out of sync with your current changes.  Sorry about that!
  - I only modified the pyzscaler/zia/admin_and_role_management.py file because, my commit contained an old version and had merge conflicts with yours.  This is why I deleted my first PR request.  To fix the issue I just copy/pasted your version into my PR.  The reason I did this is because I don't know how to cherry pick out those changes, as you are requesting.
  - I'm fine with renaming the the methods in the web_dlp.py.  I'll give it a deeper look and try again.
  -  I don't know how to do:
    - Adding the docsrc RST files to ensure our readthedocs site will pick up the changes.  -- The syntax makes no sense to me of the existing rst files.  I could use some help here.
    - Test coverage  -- I've never learned (yet) how to write pytests.  I could use some help here.

mitchos commented:
Hey @daxm I've re-pointed this pull request to the feature branch so I'll approve the pull request and just submit another one for your latest changes and we can go from there. :+1:
